### ISSUE TITLE
2022-09, Arxiv, EXPLORING VISUAL INTERPRETABILITY FOR CONTRASTIVE LANGUAGE-IMAGE PRE-TRAINING

### ISSUE THREAD

**Issue by priancho** [priancho/ResearchSurvey#16]
<img width="804" alt="image" src="https://user-images.githubusercontent.com/1505880/192962084-152b7d97-f2a0-426a-ae68-79642a1bd028.png">

<img width="725" alt="image" src="https://user-images.githubusercontent.com/1505880/192962119-197f39cd-4b41-42ab-854c-79b8dbdadf56.png">

- 이 논문은 CLIP 모델에 대한 visual interpretation 방법으로 Image-Text Similarity Map (ITSM) 이란 방법을 제안
- (참고로 classifier layer 가 없는 CLIP 과 같은 모델은 CAM 같은 방법은 사용 못함)

<img width="822" alt="image" src="https://user-images.githubusercontent.com/1505880/192962232-14ecd004-edb8-4b48-9b2a-f398722b9a88.png">

<img width="844" alt="image" src="https://user-images.githubusercontent.com/1505880/192965224-521ca2c5-0220-401c-b9b0-cbfd41d07e0c.png">

- ITSM 을 사용하여 분석한 결과, CLIP 모델은 foreground 보다 background 에 대해서 focus 를 맞추더라.
- 그 원인은 Pooling op 이더라.
- 이를 해결하기 위해서 original Attention Pooling (AP, class token) 에 추가적으로 Global Max Pooling (GMP) 를 넣었다 (dual projection)
  - 기존 projection (AP) 는 모델을 사용할 때, GMP 는 모델을 분석할 때 사용함
- 이 결과는 위의 Figure 1 (Ours) 에서 확인 가능함

<img width="831" alt="image" src="https://user-images.githubusercontent.com/1505880/192963936-fc15139a-7efe-44ba-8723-365b97457a92.png">

- ITSM (식 (1)~(5) 참조) 계산 방법은 매우 간단
- M = linear(CLIP_image(y)_1:M) @ linear(CLIP_text(x))  를 계산하고, 이를 reshape-resize-normalize 해주면 attention mask 를 만들 수 있음

<img width="824" alt="image" src="https://user-images.githubusercontent.com/1505880/192964285-0b445a3e-6139-4ba6-b522-94fd2e75cd66.png">

- (위) 실험 결과를 보면 명확함
